src/secret/types.rs

```diff
@@ -61,15 +61,43 @@ pub struct LedgerApp {
     pub(crate) version: String,
 }
 
+/// Ledger Device Type
+#[derive(Clone, Debug, Eq, PartialEq, Serialize, Deserialize)]
+pub enum LedgerDeviceType {
+    /// Device Type Nano S
+    LedgerNanoS,
+    /// Device Type Nano X
+    LedgerNanoX,
+    /// Device Type Nano S Plus
+    LedgerNanoSPlus,
+}
+
+impl From<u8> for LedgerDeviceType {
+    fn from(device: u8) -> Self {
+        match device {
+            0 => Self::LedgerNanoS,
+            1 => Self::LedgerNanoX,
+            2 => Self::LedgerNanoSPlus,
+            _ => Self::LedgerNanoS,
```

Comment:
Not a big fan of this, I'd rather impl TryFrom instead and error instead of returning a default. This may be totally fine, I'm missing the big picture, but better safe than sorry.

Reply:
I would still keep it, since it's additional info that could be helpful for someone

Reply:
Ok, but should I change it to TryFrom? Would be okay for me :man_shrugging:

Reply:
I think TryFrom would be less error prone